servant-server/test/Servant/ServerSpec.hs

```diff
@@ -667,3 +668,41 @@ errorsSpec = do
       nf <> he `shouldBe` he
       nf <> ib `shouldBe` ib
       nf <> wm `shouldBe` wm
+
+type MiscCombinatorsAPI
+  =    "version" :> HttpVersion :> Get '[JSON] String
+  :<|> "secure"  :> IsSecure :> Get '[JSON] String
+  :<|> "host"    :> RemoteHost :> Get '[JSON] String
+  :<|> "vault"   :> Vault :> Get '[JSON] ()
+
+miscApi :: Proxy MiscCombinatorsAPI
+miscApi = Proxy
+
+miscServ :: Server MiscCombinatorsAPI
+miscServ = versionHandler
+      :<|> secureHandler
+      :<|> hostHandler
+      :<|> vaultHandler
+
+  where versionHandler = return . show
+        secureHandler Secure = return "secure"
+        secureHandler NotSecure = return "not secure"
+        hostHandler = return . show
+        vaultHandler _ = return ()
```

Comment:
Well, the vault is always there, it's not like this could fail. It's just a field in the `Request`, and not a `Maybe`. But if you insist I can make it be `vaultHandler v = v `seq` return ()`.